runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/harness/WindmillStreamSender.java

```diff
@@ -57,15 +64,26 @@
 @Internal
 @ThreadSafe
 final class WindmillStreamSender implements GetWorkBudgetSpender, StreamSender {
-  private static final String STREAM_STARTER_THREAD_NAME = "StartWindmillStreamThread-%d";
-  private final AtomicBoolean started;
-  private final AtomicReference<GetWorkBudget> getWorkBudget;
-  private final GetWorkStream getWorkStream;
+  private static final Logger LOG = LoggerFactory.getLogger(WindmillStreamSender.class);
+  private static final String STREAM_MANAGER_THREAD_NAME_FORMAT = "WindmillStreamManagerThread";
+  private static final int GET_WORK_STREAM_TTL_MINUTES = 45;
+
+  private final AtomicBoolean isRunning = new AtomicBoolean(false);
   private final GetDataStream getDataStream;
   private final CommitWorkStream commitWorkStream;
   private final WorkCommitter workCommitter;
   private final StreamingEngineThrottleTimers streamingEngineThrottleTimers;
   private final ExecutorService streamStarter;
+  private final String backendWorkerToken;
+
+  @GuardedBy("activeGetWorkStream")
+  private final AtomicReference<GetWorkStream> activeGetWorkStream;
+
+  @GuardedBy("activeGetWorkStream")
+  private final AtomicReference<GetWorkBudget> getWorkBudget;
+
+  @GuardedBy("activeGetWorkStream")
```

Comment:
this doesn't seem like it needs to be guarded? any reason it has to be serially called?